pylibs/stress_tests/multicast_coaps.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020, The OTNS Authors.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the
+#    names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# CoAP Multicast Stress Test:
+#   The Border Router multicasts COAP messages to all nodes and measure the coverage and delay with injected failures.
+# Topology:
+#   Router x8 (one being Border Router)
+#   FED x8
+#   MED x8
+#   SED x8
+# Fault Injections:
+#   Packet Loss Ratio set to 0.5
+# Pass Criteria:
+#   Average Coverage >= 70%
+#   Average Delay < 200ms (2000ms for SEDs)
+#
+import os
+
+from BaseStressTest import BaseStressTest
+
+ROUTER_COUNT = 8
+FED_COUNT = 8
+MED_COUNT = 8
+SED_COUNT = 8
+
+TOTAL_NODE_COUNT = ROUTER_COUNT + FED_COUNT + MED_COUNT + SED_COUNT
+
+RADIO_RANGE = 210
+
+XMAX = 1000
+YMAX = 1000
+
+TOTAL_SIMULATION_TIME = 100 * int(os.getenv("STRESS_LEVEL", "1"))
+
+BR = None  # the Border Router
+SVR1, SVR1_DATA = "svr1", "svr1"
+BR_ADDR = 'fdde:ad00:beef:0:0:ff:fe00:fc10'
```

Comment:
How do you get this address?

Reply:
This ALOC address is fixed due the the fixed network setup process.
 - The meshlocal prefix is fixed. 
 - The service ALOC is fixed.